runtime/src/master.rs

```diff
@@ -231,13 +231,8 @@ impl<T: Trait> Module<T> {
             ensure!(valid, MasterError::<T>::BadSig);
         }
 
-        // check/execute
-        proposal
-            .clone()
-            .dispatch(system::RawOrigin::Root.into())
-            .map_err(|e| e.error)?;
-
         // execute
+        let res = proposal.clone().dispatch(system::RawOrigin::Root.into());
```

Comment:
Instead of short-circuiting, this code adds the result of the call to the `Executed` event. This is similar to pallet_sudo.

Removing the short circuit prevents the submitter from paying for errors while ensuring round_no is incremented.